File: src/main/java/net/yogstation/yogbot/listeners/channel/AbstractChannel.java
package net.yogstation.yogbot.listeners.channel;

import discord4j.common.util.Snowflake;
import discord4j.core.event.domain.message.MessageCreateEvent;
import net.yogstation.yogbot.config.DiscordChannelsConfig;
import net.yogstation.yogbot.config.DiscordConfig;
import reactor.core.publisher.Mono;

public abstract class AbstractChannel {
	protected final DiscordChannelsConfig channelsConfig;
	
	public AbstractChannel(DiscordChannelsConfig channelsConfig) {
		this.channelsConfig = channelsConfig;
	}
	
	public abstract Snowflake getChannel();
	public abstract Mono<?> handle(MessageCreateEvent event);
}

File: src/main/java/net/yogstation/yogbot/listeners/channel/AbstractChannel.kt
package net.yogstation.yogbot.listeners.channel

import discord4j.common.util.Snowflake
import discord4j.core.event.domain.message.MessageCreateEvent
import net.yogstation.yogbot.config.DiscordChannelsConfig
import reactor.core.publisher.Mono

/**
 * A class used to listen to a specific channel, handle is called when a message create event is dispatched
 * matching the channel specified in the channel property.
 */
abstract class AbstractChannel(protected val channelsConfig: DiscordChannelsConfig) {
	/**
	 * The ID of the channel to get events for
	 */
	abstract val channel: Snowflake

	/**
	 * Dispatched to handle the event
	 */
	abstract fun handle(event: MessageCreateEvent): Mono<*>
}
